mobile/native/src/db/models.rs

```diff
@@ -442,58 +455,36 @@ impl Position {
         Ok(position)
     }
 
-    // sets the position to rollover and updates the new expiry timestamp.
-    pub fn rollover(
-        conn: &mut SqliteConnection,
-        contract_symbol: ContractSymbol,
-        expiry_timestamp: OffsetDateTime,
-    ) -> Result<()> {
+    /// Update the position after the rollover protocol has started.
+    pub fn start_rollover(conn: &mut SqliteConnection, updated_position: Position) -> Result<()> {
         let affected_rows = diesel::update(positions::table)
-            .filter(schema::positions::contract_symbol.eq(contract_symbol))
-            .set((
-                positions::expiry_timestamp.eq(expiry_timestamp.unix_timestamp()),
-                positions::state.eq(PositionState::Rollover),
-                positions::updated_timestamp.eq(OffsetDateTime::now_utc().unix_timestamp()),
-            ))
+            .set(updated_position)
```

Comment:
🤔 Isn't the filter missing here? or are you arguing that there can anyways only ever be one position on the app side?

```
.filter(schema::positions::contract_symbol.eq(contract_symbol))
```

Reply:
With `AsChangeset` it will update the position matching the primary key i.e. the one with the same contract symbol.